src/main/java/org/opencraft/server/net/ActionSender.java

```diff
@@ -229,7 +229,8 @@ public void sendSpawn(
       boolean isSelf,
       boolean isVisible) {
     if (session.isExtensionSupported("ExtPlayerList", 2)) {
-      sendAddPlayerName(nameId, name, listName, teamName, (byte) 1);
+      short fixNameId = (nameId == session.getPlayer().nameId ? -1 : nameId);
```

Comment:
Same thing here as above, and then you don't have to have the weird fixNameId variable name.